networkx/generators/tests/test_random_graphs.py

```diff
@@ -5,18 +5,105 @@
 import pytest
 
 
+class TestGnpRandomGraphs:
+
+    generators = [
+        nx.gnp_random_graph,
+        nx.fast_gnp_random_graph,
+        nx.binomial_graph,
+        nx.erdos_renyi_graph,
+    ]
+
+    @pytest.mark.parametrize(
+        ("n", "p", "directed", "edges"),
+        [
+            (10, -1.1, False, 0),
+            (10, 1.1, False, 45),
+            (10, -1.1, True, 0),
+            (10, 1.1, True, 90),
+        ],
+    )
+    def test_basic_properties(self, n, p, directed, edges):
+        for generator in self.generators:
+            G = generator(n, p, directed=directed)
```

Comment:
Instead of looping over the generators, the tests can be parametrized over them as well! This can be accomplished by stacking the parametrization decorators - see [the pytest docs](https://docs.pytest.org/en/6.2.x/parametrize.html#parametrizing-fixtures-and-test-functions) for more info.